File: growie_app/api/videos.py
import frappe


@frappe.whitelist()
def get_videos(limit=50):
    videos = frappe.get_all(
        "Videos",
        fields=["name", "title", "category", "related_video", "youtube_url", "whats_all_about", "creation"],
        order_by="creation desc",
        limit=int(limit),
    )

    result = []
    for v in videos:
        result.append({
            "id": v.name,
            "title": v.title or "",
            "description": v.whats_all_about or "",
            "category": v.category or "education",
            "videoUrl": v.related_video or "",
            "youtubeUrl": v.youtube_url or "",
            "publishedAt": str(v.creation),
        })

    return result

File: frontend/services/videos.ts
export interface VideoRecord {
  id: string
  title: string
  description: string
  category: string
  videoUrl: string
  youtubeUrl: string
  publishedAt: string
}

export async function getVideos(limit = 50): Promise<VideoRecord[]> {
  const params = new URLSearchParams({ limit: String(limit) })
  const response = await fetch(
    `/api/method/growie_app.api.videos.get_videos?${params.toString()}`,
    { credentials: 'include', headers: { Accept: 'application/json' } }
  )
  const data = await response.json()
  return Array.isArray(data?.message) ? (data.message as VideoRecord[]) : []
}